src/alasqlparser.jison

```diff
@@ -1460,9 +1460,43 @@ LogicValue
 
 StringValue
 	: STRING
-		{ $$ = new yy.StringValue({value: $1.substr(1,$1.length-2).replace(/(\\\')/g,"'").replace(/(\'\')/g,"'")}); }
+		{ 
+			var str = $1.substr(1,$1.length-2).replace(/(\'\')/g,"'");
+			str = str.replace(/\\(.)/g, function(match, char) {
```

Comment:
You dont know if the first \ is escaped beforehand. You will need to find pairs like \\ (regex as \\\\) or not pair (what you got now \\(.))

Please see if you can place regex outside of the executing function